### ISSUE TITLE
Training with Sparsity

### ISSUE THREAD

**Issue by MissyDu** [Eric-mingjie/network-slimming#11]
This process can be fine-tuned with a "Normal Trained"  model?

I prune InceptionV3 by fine-tuning with a pre-trained ImageNet model, and find it slow to be sparse(base_lr=0.01, 30epoch). I wonder whether if it is possible to use fine-tuning.

And how do you think InceptionV3 which has many branchs? I find it's a trouble to implement InceptionV3prune.py.

**Comment by Eric-mingjie** [comment]
It is better to train from start with sparsity. We haven't tried fine-tuning with sparsity on a pretrained model before.

To implement pruning, you just need to collect all the scaling factors in bn layers. Use `model.modules()` to go through all the modules.

**Comment by MissyDu** [comment]
@Eric-mingjie , always thanks for quick answer.
I have implemented InceptionV3prune.py, but found accuracy drops a lot after prune.
The trouble of InceptionV3prune.py is that it has many branchs. So it means the cfg_mask can't be read one by one in turn. My main modification is to solve this problem.

**Comment by Eric-mingjie** [comment]
The significant drop in accuracy could be due to that the model is not sparse. You can analyze the magnitude of scaling factors in the models.

**Comment by MissyDu** [comment]
@Eric-mingjie ,yes. The model is not sparse enough.
One more question,  druning the sparsity training, the channel_selection is necessary?

**Comment by Eric-mingjie** [comment]
The use of `channel_selection` depends on the architecture you use and the way you want to do pruning. For VGG, no `channel selection` is needed. For ResNet and DenseNet, `channel selection` is normally needed.

However, if you use [mask implementation](https://github.com/Eric-mingjie/network-slimming/tree/master/mask-impl), `channel selection` is not needed for any model.
